### PR TITLE
[CUBRIDQA-864] add shell_long test Guide

### PR DESCRIPTION
**Pull request by slamdunkorchid**
refer to http://jira.cubrid.org/browse/CUBRIDQA-864

### CODE REVIEW

File: doc/shell_long/README.md [hunk 1]
```diff
@@ -0,0 +1,891 @@
+# 1. Test Objective  
+The document is to introduce how to test shell_long via CTP. It is similiar with shell test with CTP.For some shell cases, it spend so much time which affects the end of shell regression test, then remove them to shell_long test category.    
+
+# 2. Tools Introduction  
+## 2.1 CTP tools
+Please refer to [README.md](https://github.com/CUBRID/cubrid-testtools/blob/develop/CTP/README.md) of CTP
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Need inherit shell guide (provide a link).

General usage should be merged to shell guide.

Please check information provided in this guide and decide whether to merge to shell guide.

File: doc/shell_long/README.md [hunk 2]
```diff
@@ -0,0 +1,891 @@
+# 1. Test Objective  
+The document is to introduce how to test shell_long via CTP. It is similiar with shell test with CTP.For some shell cases, it spend so much time which affects the end of shell regression test, then remove them to shell_long test category.
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]:
```suggestion
This guide is to introduce how to execute shell_long test suite via test tool CTP.  The shell_long test case specification is exactly same as general SHELL test case's. The only difference between them is that shell_long test cases ask a little long time to execute. 

```

File: doc/shell_long/README.md [hunk 3]
```diff
@@ -0,0 +1,891 @@
+# 1. Test Objective  
+The document is to introduce how to test shell_long via CTP. It is similiar with shell test with CTP.For some shell cases, it spend so much time which affects the end of shell regression test, then remove them to shell_long test category.    
+
+# 2. Tools Introduction
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: As is:

# 2. Tools Introduction  

To be:

# 2. Test Introduction  
2.1 CTP Test Tool
2.2 Shell_long Test Cases

File: doc/shell_long/README.md [hunk 4]
```diff
@@ -0,0 +1,891 @@
...
+```
+3. checkout git repositories to $HOME/
+```
+git clone https://github.com/CUBRID/cubrid-testtools.git
+cd ~/cubrid-testtools/
+git checkout develop
+```
+### On Controller node
+1. Install CTP  
+Please follow [the guide to install CTP](http://jira.cubrid.org/browse/CUBRIDQA-835?focusedCommentId=4753733&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4753733).
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Find CTP installation link in https://github.com/CUBRID/cubrid-testtools

File: doc/shell_long/README.md [hunk 5]
```diff
@@ -0,0 +1,891 @@
...
+compile C code on linux or windows platform.  
+```
+xgcc -I${CUBRID}/include -L${CUBRID}/lib -lcascci ${MODE}  -o test2 test2.c
+xgcc -o cci cci.c -lpthread
+xgcc -g -o testbug9251 sus9251.c 
+xgcc -o test test.c
+```
+
+
+**Notice: there are other common functions,you can refer to $init_path/init.sh**
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Link to shell guide which describe this.

File: doc/shell_long/README.md [hunk 6]
```diff
@@ -0,0 +1,891 @@
...
+AUTO_10.2.0.8060-689ccdd_20181019_014026.tar.gz  AUTO_10.2.0.8329-51e235e_20190413_151229.tar.gz               AUTO_SERVER-START_10.2.0.8362-fbf9d84_20190526_055005.tar.gz
+AUTO_10.2.0.8060-689ccdd_20181019_040434.tar.gz  AUTO_10.2.0.8349-bb21e2d_20190428_190725.tar.gz               AUTO_SERVER-START_10.2.0.8362-fbf9d84_20190526_055233.tar.gz
+AUTO_10.2.0.8107-a05cfaa_20181028_053656.tar.gz  AUTO_10.2.0.8362-fbf9d84_20190526_060734.tar.gz
+AUTO_10.2.0.8254-c015eb2_20190213_025432.tar.gz  AUTO_SERVER-START_10.2.0.8254-c015eb2_20190213_020024.tar.gz
+```  
+# 5.Test Case Specification
+## 5.1 Requirements
+export init_path=$HOME/CTP/shell/init_path
+
+## 5.2 How To Write Test Case
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Check and merge to shell guide (written by Mandy)

File: doc/shell_long/README.md [hunk 7]
```diff
@@ -0,0 +1,635 @@
...
+This guide is to introduce how to execute shell_long test suite via test tool CTP.  The shell_long test case specification is exactly same as general SHELL test case's. The only difference between them is that shell_long test cases ask a little long time to execute.     
+
+# 2. Test Introduction  
+## 2.1 CTP test tool
+Please refer to [ctp usage for shell](https://github.com/CUBRID/cubrid-testtools/blob/develop/doc/shell/shell_guide.md#2-ctp-introduction)   
+the CTP usage for shell_long regression test is as below   
+```bash
+ctp.sh shell -c ~/CTP/conf/shell_runtime.conf
+```
+for regression test, we will copy `shell_template_for_shell_long.conf` to `shell_runtime.conf`
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Follow same way to describe in in cci guide.
It's better not to describe things related to regression test. We hope that developer users may get guide from here.

File: doc/shell_long/README.md [hunk 8]
```diff
@@ -0,0 +1,635 @@
+# 1. Test Objective  
+This guide is to introduce how to execute shell_long test suite via test tool CTP.  The shell_long test case specification is exactly same as general SHELL test case's. The only difference between them is that shell_long test cases ask a little long time to execute.     
+
+# 2. Test Introduction  
+## 2.1 CTP test tool
+Please refer to [ctp usage for shell](https://github.com/CUBRID/cubrid-testtools/blob/develop/doc/shell/shell_guide.md#2-ctp-introduction)
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Correct to the link for common installation guide of CTP.

File: doc/shell_long/README.md [hunk 9]
```diff
@@ -0,0 +1,635 @@
...
+...
+./2hour/_02_cursor_stress/cases/_02_cursor_stress.sh
+./2hour/bug_bts_10009/cases/bug_bts_10009.sh
+./2hour/bug_bts_9969/cases/bug_bts_9969.sh
+```
+How to write new shell_long test cases   
+please see [shell long cases specification](#5test-case-specification)
+
+# 3. Deploy Shell_long Regression Test
+In this section, we will introduce how to deploy the Shell_long Regression Test Environment.
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]:
```suggestion
In this section, we will introduce how to deploy the shell_long Regression Test Environment.  
```

File: doc/shell_long/README.md [hunk 10]
```diff
@@ -0,0 +1,635 @@
...
+```
+3. checkout git repositories to $HOME/
+```
+git clone https://github.com/CUBRID/cubrid-testtools.git
+cd ~/cubrid-testtools/
+git checkout develop
+```
+### On Controller node  
+1. Install CTP      
+Please follow [the guide to install CTP](https://github.com/CUBRID/cubrid-testtools/blob/develop/doc/ctp_install_guide.md).
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Change link to https://github.com/CUBRID/cubrid-testtools/blob/develop/doc/ctp_install_guide.md#3-install-ctp-as-regression-test-platform

File: doc/shell_long/README.md [hunk 11]
```diff
@@ -0,0 +1,635 @@
...
+```
+ 
+5. start consumer process  
+```bash
+ sh start_test.sh
+```
+
+### On Test nodes
+1. Install CTP  
+Please follow [the guide to install CTP](https://github.com/CUBRID/cubrid-testtools/blob/develop/doc/ctp_install_guide.md).
```
Comment (fanzaiqiang) [inline doc/shell_long/README.md]: Change the link as above request.

**Comment by fanzaiqiang** [comment]
Please refine format according to review face to face.

**Comment by slamdunkorchid** [comment]
Please continue to review

**Comment by fanzaiqiang** [comment]
Move guide files to `~/doc` folder and refresh image links.
Then merge codes.

**Pull request merged**